src/webview/panel-views/debuggerPanel.ts

```diff
@@ -0,0 +1,350 @@
+        dummy: undefined,
+      };
+    });
+  }
+
+  private stopGifRecording(): void {
+    this.stopRecording<'gif'>('gif', (handler) => {
+      //  TODO: VOOD-171: Stop GIF recording
+      console.warn('DebuggerPanel.stopGifRecording', 'handler', handler);
+    });
+  }
+
+  private startRecording<Format extends RecordingFormat>(
+    format: RecordingFormat,
+    contextFactory: (canvas: HTMLCanvasElement) => RecordingContext<Format>
+  ): void {
+    if (this.recordingContextsByFormat[format]) {
+      console.warn(`${format} recording already in progress`);
+    } else {
+      const { canvas } = this;
+      if (canvas) {
+        (this.recordingContextsByFormat[format] as RecordingContext<Format>) = contextFactory(canvas);
+      } else {
+        console.warn(`Cannot start ${format} recording without canvas`);
+      }
+    }
+    this.updateRecordingIndicatorVisibility();
+  }
+
+  private stopRecording<Format extends RecordingFormat>(
+    format: RecordingFormat,
+    contextHandler: (context: RecordingContext<Format>) => void
+  ): void {
+    const context = this.recordingContextsByFormat[format] as RecordingContext<Format>;
+    if (context) {
+      contextHandler(context);
+      delete this.recordingContextsByFormat[format];
+    } else {
+      console.warn(`No ${format} recording in progress`);
+    }
+    this.updateRecordingIndicatorVisibility();
+  }
+
+  private updateRecordingIndicatorVisibility(): void {
+    this.recordingIndicator.style.visibility = Object.keys(this.recordingContextsByFormat).length > 0 ? 'visible' : 'hidden';
+  }
+
+  private resetCanvas(): void {
+    this.canvas = this.document.querySelector(`.${renderingAreaClassName} canvas`);
+  }
+
+  private enableToolBarControls(): void {
+    this.toolBarControls.forEach((control) => (control.disabled = false));
+  }
+
+  private get toolBarControls(): (Button | Dropdown)[] {
+    return [...this.toolBarButtons, this.stackFramesDropdown];
+  }
+
+  private downloadBlobFile(fileName: string, blob: Blob): void {
+    const url = URL.createObjectURL(blob);
+    this.downloadFile(fileName, url);
+    URL.revokeObjectURL(url);
+  }
+
+  private downloadFile(fileName: string, url: string): void {
+    const link = this.document.createElement('a');
+    link.download = fileName;
+    link.href = url;
+    link.click();
+  }
+
+  protected abstract openAllClusters(): void;
+
+  //  Window event handling
+
+  private initializeMessageHandling(): void {
+    this.window.addEventListener(
+      'message',
+      ({ data }: MessageEvent<PanelViewProxyMessage<RenderingAreaData, RenderingAreaOptions, RenderingAreaUpdateData>>) =>
+        this.onPanelViewProxyMessageReceived(data)
+    );
+  }
+
+  private onPanelViewProxyMessageReceived(
+    message: DebuggerPanelMessage | PanelViewProxyMessage<RenderingAreaData, RenderingAreaOptions, RenderingAreaUpdateData>
+  ): void {
+    console.debug('DebuggerPanel.onPanelViewProxyMessageReceived', 'message', message);
```

Comment:
```suggestion
```

@GinoCardillo-OST Should I remove this line as well?

Reply:
yes, please

Reply:
Done.